src/iop/finalscale.c

```diff
@@ -148,6 +150,22 @@ void cleanup(dt_iop_module_t *self)
   self->default_params = NULL;
 }
 
+typedef struct dt_iop_finalscale_gui_data_t
+{ } dt_iop_finalscale_gui_data_t;
+
+dt_iop_finalscale_gui_data_t dummy;
+
+void gui_init(dt_iop_module_t *self)
+{
+  IOP_GUI_ALLOC(finalscale);
+  self->widget = gtk_label_new(NULL);
+  gtk_label_set_markup(GTK_LABEL(self->widget),_("This module is used to downscale images at export time. "
```

Comment:
I think this is plain wrong.

Reply:
We should in any case remove all capital letters to keep standard dt style.

Reply:
@jenshannoschwalm : Plain wrong? You mean that `export time` is confusing? Because to me this module does clip/zoom to final size looking at the code. This module is indeed used for all pixelpipe.

Reply:
capitals removed

-d pipe indicates the existence of finalscale just on export if high quality resampling is active:
```
   299,0892 [modify roi OUT]             [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   299,0892 [modify roi IN]              [export]       finalscale           (   0/   0) 5202x3460 scale=1,0000 --> (   0/   0)  648x 431 scale=0,1246
   299,0892 [modify roi IN]              [export]       highlights           (   0/   0) 5202x3464 scale=1,0000 --> (   0/   0) 5202x3460 scale=1,0000
   299,0892 [modify roi IN]              [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   299,0927 [pixelpipe_process_CL]       [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
...
   299,3029 [pixelpipe_process_CL]       [export]       colorout             (   0/   0) 5202x3460 scale=1,0000 --> (   0/   0) 5202x3460 scale=1,0000
   299,3447 [pixelpipe_process_CL]       [export]       finalscale           (   0/   0) 5202x3460 scale=1,0000 --> (   0/   0)  648x 431 scale=0,1246
   299,3448 [clip_and_zoom_roi CL]       [export]       finalscale           (   0/   0) 5202x3460 scale=1,0000 --> (   0/   0)  648x 431 scale=0,1246 device=0
   299,3584 [cache report]               [export]                             2 lines (important=0, used=2). Used 575MB, limit=0MB. Hitrate=0,00
```

export no high quality resampling:
```
   292,7390 [modify roi OUT]             [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   292,7390 [modify roi IN]              [export]       demosaic             (   0/   0) 5202x3459 scale=1,0000 --> (   0/   0)  648x 431 scale=0,1246
   292,7391 [modify roi IN]              [export]       highlights           (   0/   0) 5202x3464 scale=1,0000 --> (   0/   0) 5202x3459 scale=1,0000
   292,7391 [modify roi IN]              [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   292,7434 [pixelpipe_process_CL]       [export]       rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
...
   292,8721 [pixelpipe_process_CL]       [export]       colorout             (   0/   0)  648x 431 scale=0,1246 --> (   0/   0)  648x 431 scale=0,1246
   292,8742 [pixelpipe_process_on_CPU]   [export]       gamma                (   0/   0)  648x 431 scale=0,1246 --> (   0/   0)  648x 431 scale=0,1246
   292,8745 [cache report]               [export]                             2 lines (important=1, used=2). Used 575MB, limit=0MB. Hitrate=0,00

```
full pipe:      
```
   274,7225 [modify roi OUT]             [full]         rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   274,7225 [pixelpipe_cache_checkmem]   [full]                               64 lines (important=0, used=0). Cache: freed=0MB (bad=0 low=0 high=0). Now using 0MB, limit=512MB
   274,7225 [modify roi IN]              [full]         demosaic             (   0/   0) 5202x3462 scale=1,0000 --> (   0/   0) 2096x1395 scale=0,4029
   274,7225 [modify roi IN]              [full]         highlights           (   0/   0) 5202x3464 scale=1,0000 --> (   0/   0) 5202x3462 scale=1,0000
   274,7226 [modify roi IN]              [full]         rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
   274,7262 [pixelpipe_process_CL]       [full]         rawprepare           (   0/   0) 5360x3516 scale=1,0000 --> (   0/   0) 5202x3464 scale=1,0000
...
   274,8739 [pixelpipe_process_CL]       [full]         colorout             (   0/   0) 2096x1395 scale=0,4029 --> (   0/   0) 2096x1395 scale=0,4029
   274,8826 [pixelpipe_process_on_CPU]   [full]         gamma                (   0/   0) 2096x1395 scale=0,4029 --> (   0/   0) 2096x1395 scale=0,4029
   274,8869 [cache report]               [full]                               64 lines (important=2, used=10). Used 518MB, limit=512MB. Hitrate=0,00```

so the phrase "used to downscale images at export time" might not be that precise but not wrong.
maybe better " ...used to downscale images with high quality resampling"

Reply:
Indeed, you're right and this is new to me. The module finalscale is only used for high quality export.

Reply:
Yes, but also for upscaling